File: e2e/node/tests/map.test.ts
import { describe, it, expect } from "vitest";
import { scrape, createEngine } from "@kreuzberg/kreuzcrawl";

describe("map", () => {
	it("map_discover_urls: Discovers all URLs on a site without fetching full content", async () => {
		const engine = createEngine(null);
		const url = process.env.MOCK_SERVER_URL + "/fixtures/map_discover_urls";
		await scrape(engine, url);
		// skipped: field 'urls.length' not available on result type
	});

	it("map_exclude_patterns: Excludes URLs matching patterns from URL map", async () => {
		const engine = createEngine(null);
		const url = process.env.MOCK_SERVER_URL + "/fixtures/map_exclude_patterns";
		await scrape(engine, url);
		// skipped: field 'urls.length' not available on result type
	});

	it("map_include_subdomains: Includes subdomain URLs in URL map discovery", async () => {
		const engine = createEngine(null);
		const url = process.env.MOCK_SERVER_URL + "/fixtures/map_include_subdomains";
		await scrape(engine, url);
		// skipped: field 'urls.length' not available on result type
		// skipped: field 'urls' not available on result type
	});

	it("map_large_sitemap: Handles large sitemap with 100+ URLs", async () => {
		const engine = createEngine(null);
		const url = process.env.MOCK_SERVER_URL + "/fixtures/map_large_sitemap";
		await scrape(engine, url);
		// skipped: field 'urls.length' not available on result type
	});

	it("map_limit_pagination: Limits map result count to specified maximum", async () => {
		const engine = createEngine(null);
		const url = process.env.MOCK_SERVER_URL + "/fixtures/map_limit_pagination";
		await scrape(engine, url);
		// skipped: field 'urls.length' not available on result type
	});

	it("map_search_filter: Filters map results by search keyword", async () => {
		const engine = createEngine(null);
		const url = process.env.MOCK_SERVER_URL + "/fixtures/map_search_filter";
		await scrape(engine, url);
		// skipped: field 'urls.length' not available on result type
		// skipped: field 'urls' not available on result type
	});
});

File: e2e/rust/tests/map_test.rs
//! E2e tests for category: map

use kreuzcrawl::scrape;
use kreuzcrawl::create_engine;

#[tokio::test]
async fn test_map_discover_urls() {
    // Discovers all URLs on a site without fetching full content
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = format!("{}/fixtures/{}", std::env::var("MOCK_SERVER_URL").expect("MOCK_SERVER_URL not set"), "map_discover_urls");
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'urls.length' not available on result type
}

#[tokio::test]
async fn test_map_exclude_patterns() {
    // Excludes URLs matching patterns from URL map
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = format!("{}/fixtures/{}", std::env::var("MOCK_SERVER_URL").expect("MOCK_SERVER_URL not set"), "map_exclude_patterns");
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'urls.length' not available on result type
}

#[tokio::test]
async fn test_map_include_subdomains() {
    // Includes subdomain URLs in URL map discovery
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = format!("{}/fixtures/{}", std::env::var("MOCK_SERVER_URL").expect("MOCK_SERVER_URL not set"), "map_include_subdomains");
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'urls.length' not available on result type
    // skipped: field 'urls' not available on result type
}

#[tokio::test]
async fn test_map_large_sitemap() {
    // Handles large sitemap with 100+ URLs
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = format!("{}/fixtures/{}", std::env::var("MOCK_SERVER_URL").expect("MOCK_SERVER_URL not set"), "map_large_sitemap");
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'urls.length' not available on result type
}

#[tokio::test]
async fn test_map_limit_pagination() {
    // Limits map result count to specified maximum
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = format!("{}/fixtures/{}", std::env::var("MOCK_SERVER_URL").expect("MOCK_SERVER_URL not set"), "map_limit_pagination");
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'urls.length' not available on result type
}

#[tokio::test]
async fn test_map_search_filter() {
    // Filters map results by search keyword
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = format!("{}/fixtures/{}", std::env::var("MOCK_SERVER_URL").expect("MOCK_SERVER_URL not set"), "map_search_filter");
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'urls.length' not available on result type
    // skipped: field 'urls' not available on result type
}
